### PR TITLE
Add obs package status

### PR DESCRIPTION
**Pull request by hennevogel**

### CODE REVIEW

**Comment by hennevogel** [comment]
@DavidKang continued to refactor a bit to make this more generic, please review

File: obs-package-status/status-check.rb [hunk 1]
```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/ruby
+require 'net/http'
+require 'nokogiri'
+require 'json'
+# require 'pry'
+require 'dotenv/load'
+
+# OBS Configuration
+OBS_API_URL = ENV['OBS_API_URL'] || 'https://api.opensuse.org'
```
Comment (dmarcoux) [inline obs-package-status/status-check.rb]: I like using this form for default values: `ENV.fetch('OBS_API_URL', 'https://api.opensuse.org')`

File: obs-package-status/dotenv.example [hunk 2]
```diff
@@ -0,0 +1,10 @@
+OBS_TRELLO_API_KEY=12345
```
Comment (dmarcoux) [inline obs-package-status/dotenv.example]: I believe this file should be named `.env.example`

File: obs-package-status/README.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
+# OBS Package Status Scanner
+This script shows if the packages of OBS (Open Build Service) are broken or not. The status gets then reported to a Trello card.
+
+## Configuration
+This bot is configured via environment variables. Either set them however you set them
+or copy the `dotenv.example` file to `.env` and the script picks them up from this file.
```
Comment (dmarcoux) [inline obs-package-status/README.md]: `dotenv.example` should be renamed to `.env.example`

File: obs-package-status/Gemfile [hunk 4]
```diff
@@ -0,0 +1,5 @@
+source 'https://rubygems.org'
+
+gem 'nokogiri'
+gem 'pry'
```
Comment (DavidKang) [inline obs-package-status/Gemfile]: I think we can remove this gem, or is for the future if we need to debug?

File: obs-package-status/status-check.rb [hunk 5]
```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/ruby
+require 'net/http'
+require 'nokogiri'
+require 'json'
+# require 'pry'
```
Comment (DavidKang) [inline obs-package-status/status-check.rb]: If we remove the gem we also need remove this comment

File: obs-package-status/status-check.rb [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+    unless response.is_a?(Net::HTTPOK)
+      abort(response.message)
+    end
+
+    status = Nokogiri::XML(response.body).css('status')
+
+    # Bug: openSUSE/open-build-service#6924
+    if status.empty?
+      abort("Can't find architecture #{OBS_ARCHITECTURE}")
```
Comment (DavidKang) [inline obs-package-status/status-check.rb]: [nitpick] This could be written in one line.

File: obs-package-status/status-check.rb [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+  request.set_form_data({
+    "value" => trello_card_content
+  })
+  Net::HTTP.start(card_desc_uri.hostname, card_desc_uri.port, use_ssl: true) { |http| http.request(request) }
+end
+
+
+def change_trello_cover(status:)
+  image_name = status ? TRELLO_PASSED_IMAGE : TRELLO_FAILED_IMAGE
+  cover_id = get_trello_image_id(image_name: image_name)
```
Comment (DavidKang) [inline obs-package-status/status-check.rb]: :star2:

File: obs-package-status/status-check.rb [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+    puts "Warning: can't find image with name #{image_name}, not changing card cover"
+    return
+  end
+
+  cover_uri = URI("https://api.trello.com/1/cards/#{TRELLO_CARD_ID}/idAttachmentCover?value=#{cover_id}&key=#{TRELLO_KEY}&token=#{TRELLO_TOKEN}")
+  request = Net::HTTP::Put.new(cover_uri)
+  response = Net::HTTP.start(cover_uri.hostname, cover_uri.port, use_ssl: true) { |http| http.request(request) }
+
+  unless response.is_a?(Net::HTTPOK)
+    abort(response.message)
```
Comment (DavidKang) [inline obs-package-status/status-check.rb]: [nitpick] This could be written in one line

File: obs-package-status/status-check.rb [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+  end
+end
+
+def get_trello_image_id(image_name:)
+  uri = URI("https://api.trello.com/1/cards/#{TRELLO_CARD_ID}/attachments?key=#{TRELLO_KEY}&token=#{TRELLO_TOKEN}")
+  response = Net::HTTP.get_response(uri)
+
+  unless response.is_a?(Net::HTTPOK)
+    abort(response.message)
```
Comment (DavidKang) [inline obs-package-status/status-check.rb]: [nitpick] This could be written in one line

File: obs-package-status/Gemfile (hunk 4 above)
Comment (hennevogel) [reply obs-package-status/Gemfile]: yeah I don't want to install it manually all the time...

File: obs-package-status/README.md (hunk 3 above)
Comment (hennevogel) [reply obs-package-status/README.md]: Not sure a 'hidden' file in the repo this is too obvious...

**Pull request merged**